app/scripts/shell/dimSearchFilter.directive.js

```diff
@@ -110,15 +111,21 @@
       }
 
       filterFn = function(item) {
-        return (_.reduce(tempFns, function(memo, fn) {
-          return memo || fn(item);
-        }, false));
+        var checks = 0;
+        _.each(filters, function(filter){
+          if(item.hasOwnProperty([filter.predicate])){
```

Comment:
Where is the check on the tempFns?  Those functions need to be executed to determine if an item should be visible.  A simple property check isn't sufficient for things like incomplete.  You'll have to test the item.